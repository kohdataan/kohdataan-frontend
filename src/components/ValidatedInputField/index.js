import React, { memo } from 'react'
import PropTypes from 'prop-types'

const ValidatedInputField = React.forwardRef((props, ref) => {
  const {
    label,
    name,
    showLabel,
    showPlaceholder,
    onChange,
    ariaInvalid,
    inputClassName,
    labelClassName,
    type,
  } = props
  return (
    <label htmlFor={label} className={inputClassName}>
      {showLabel && label}
      <input
        type={type}
        name={name}
        ref={ref}
        id={label}
        onChange={onChange}
        aria-label={label}
        aria-invalid={ariaInvalid}
        placeholder={showPlaceholder ? label : ''}
        className={labelClassName}
      />
    </label>
  )
})

ValidatedInputField.propTypes = {
  type: PropTypes.string,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  showLabel: PropTypes.bool,
  showPlaceholder: PropTypes.bool,
  ariaInvalid: PropTypes.bool,
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
}

ValidatedInputField.defaultProps = {
  type: 'text',
  showLabel: true,
  showPlaceholder: true,
  ariaInvalid: false,
  inputClassName: '',
  labelClassName: '',
  onChange: null,
}

export default memo(ValidatedInputField)
