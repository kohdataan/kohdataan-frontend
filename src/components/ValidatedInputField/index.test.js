import React from 'react'
import { create } from 'react-test-renderer'
import ValidatedInputField from './index'

describe('Button container', () => {
  test('it matches the snapshot', () => {
    const component = create(<ValidatedInputField label="test" name="test" />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
