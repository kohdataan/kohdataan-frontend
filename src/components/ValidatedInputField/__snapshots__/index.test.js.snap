// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button container it matches the snapshot 1`] = `
<label
  className=""
  htmlFor="test"
>
  test
  <input
    aria-invalid={false}
    aria-label="test"
    className=""
    disabled={false}
    id="test"
    name="test"
    onChange={null}
    placeholder="test"
    type="text"
  />
</label>
`;
